pkg/api/azenvtypes.go

```diff
@@ -7,60 +7,60 @@ import "fmt"
 
 //AzureEnvironmentSpecConfig is the overall configuration differences in different cloud environments.
 type AzureEnvironmentSpecConfig struct {
-	CloudName            string
-	DockerSpecConfig     DockerSpecConfig
-	KubernetesSpecConfig KubernetesSpecConfig
-	DCOSSpecConfig       DCOSSpecConfig
-	EndpointConfig       AzureEndpointConfig
-	OSImageConfig        map[Distro]AzureOSImageConfig
+	CloudName            string                        `json:"cloudName,omitempty"`
+	DockerSpecConfig     DockerSpecConfig              `json:"dockerSpecConfig,omitempty"`
+	KubernetesSpecConfig KubernetesSpecConfig          `json:"kubernetesSpecConfig,omitempty"`
+	DCOSSpecConfig       DCOSSpecConfig                `json:"dcosSpecConfig,omitempty"`
+	EndpointConfig       AzureEndpointConfig           `json:"endpointConfig,omitempty"`
+	OSImageConfig        map[Distro]AzureOSImageConfig `json:"osImageConfig,omitempty"`
 }
 
 //DockerSpecConfig is the configurations of docker
 type DockerSpecConfig struct {
-	DockerEngineRepo         string
-	DockerComposeDownloadURL string
+	DockerEngineRepo         string `json:"dockerEngineRepo,omitempty"`
+	DockerComposeDownloadURL string `json:"dockerComposeDownloadURL,omitempty"`
 }
 
 //DCOSSpecConfig is the configurations of DCOS
 type DCOSSpecConfig struct {
-	DCOS188BootstrapDownloadURL     string
-	DCOS190BootstrapDownloadURL     string
-	DCOS198BootstrapDownloadURL     string
-	DCOS110BootstrapDownloadURL     string
-	DCOS111BootstrapDownloadURL     string
-	DCOSWindowsBootstrapDownloadURL string
-	DcosRepositoryURL               string // For custom install, for example CI, need these three addributes
-	DcosClusterPackageListID        string // the id of the package list file
-	DcosProviderPackageID           string // the id of the dcos-provider-xxx package
+	DCOS188BootstrapDownloadURL     string `json:"dcos188BootstrapDownloadURL,omitempty"`
```

Comment:
Is there any way we can avoid updating the DCOS surface area here? Does the way we're composing types require this?

Reply:
We will only support k8s on AzureStack, I will remove that.